src/modules/commander/ManualControl.hpp

```diff
@@ -0,0 +1,101 @@
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file ManualControl.hpp
+ *
+ * @brief Logic for handling RC or Joystick input
+ *
+ * @author Matthias Grob <maetugr@gmail.com>
+ */
+
+#pragma once
+
+#include <lib/hysteresis/hysteresis.h>
+#include <px4_platform_common/module_params.h>
+
+#include <uORB/Subscription.hpp>
+#include <uORB/topics/manual_control_setpoint.h>
+#include <uORB/topics/manual_control_switches.h>
+#include <uORB/topics/vehicle_control_mode.h>
+#include <uORB/topics/vehicle_status.h>
+
+class ManualControl : ModuleParams
+{
+public:
+	ManualControl(ModuleParams *parent) : ModuleParams(parent) {};
+	~ManualControl() override = default;
+
+	void setRCAllowed(const bool rc_allowed) { _rc_allowed = rc_allowed; }
+
+	/**
+	 * Check for manual control input changes and process them
+	 * @return true if there was new data
+	 */
+	bool update();
+	bool isRCAvailable() { return _rc_available; }
+	bool wantsOverride(const vehicle_control_mode_s &vehicle_control_mode);
+	bool wantsDisarm(const vehicle_control_mode_s &vehicle_control_mode, const vehicle_status_s &vehicle_status,
+			 manual_control_switches_s &manual_control_switches, const bool landed);
+	bool wantsArm(const vehicle_control_mode_s &vehicle_control_mode, const vehicle_status_s &vehicle_status,
+		      manual_control_switches_s &manual_control_switches, const bool landed);
+	bool isThrottleLow() { return _last_manual_control_setpoint.z < 0.1f; }
+	bool isThrottleAboveCenter() { return _last_manual_control_setpoint.z > 0.6f; }
+	hrt_abstime getLastRCTimestamp() { return _last_manual_control_setpoint.timestamp; }
```

Comment:
Just a nitpick (no need to change it here, but I want to share my view because I just discussed it with @julianoes). For a long time, I never knew if I should write `GPS` vs `Gps`, or `getLastRcTimestamp` vs `getLastRcTimestamp`. Working on an auto-generation system, I realized that the latter was always better.

I see it like this: `getLastRcTimestamp` is exactly equivalent to `get_last_rc_timestamp`, but `getLastRCTimestamp` requires vocabulary knowledge (is it `get_last_r_c_timestamp`, or maybe `get_last_r_ctimestamp`?). What if one day you want to parse the name?

And to give a real example: the Swift version of protobuf written by Apple tries to go the aesthetic way, and uses `HTTP`, `ID`, and exactly 2 others I don't remember. But that requires a vocabulary of acronyms, and since they don't know `RC`, it has to be `Rc`. Worse, they know `HTTP`, but they use `Ftp` (how consistent, eh?). Even worse: they have extra code handling those 4 exceptions (they need to go change `Http` to `HTTP` and `Id` to `ID` everywhere), and they have extra unit tests for that.

TL;DR: Camelcase says that each word starts with a capital letter. `GPS` is a word, so in camelcase, it becomes `Gps`.

Reply:
You are right! I honestly didn't think that far and just wrote it how I considered it most readable but I don't see any disadvantage in terms of understanding with pure camel case but an advantage in consistency 👍

Reply:
d909125